### PR TITLE
Add the hit block event value to the on projectile hit: event

### PR DESCRIPTION
**Pull request by DeltaRays**
### Description
(and added another example for entity dismounting to SimpleEvents and reformatted SimpleEvents)

---
**Target Minecraft Versions:** 1.11+     
**Requirements:**    none 
**Related Issues:**    none

### CODE REVIEW

**Comment by Wealthyturtle** [comment]
The SimpleEvents reformatting should be a separate PR

**Comment by DeltaRays** [comment]
Oops

**Comment by Pikachu920** [comment]
it looks like the formatting changes are still there

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 1]
```diff
@@ -440,6 +440,13 @@ public DamageCause get(final EntityDeathEvent e) {
 			}
 		}, 0);
 		// ProjectileHitEvent
+		EventValues.registerEventValue(ProjectileHitEvent.class, Block.class, new Getter<Block, ProjectileHitEvent>() {
+			@Nullable
+			@Override
+			public Block get(ProjectileHitEvent e) {
+				return e.getHitBlock();
+			}
+		}, 0);
```
Comment (FranKusmiruk) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:449]: Needs a version check, this didn't method didn't exist till 1.11

Comment (DeltaRays) [reply src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:449]: Sorry, how would I add that?

Comment (DeltaRays) [reply src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:449]: I changed the comment to 1.11+, is there anything else I need to do?

Comment (ThatOneTqnk) [reply src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:449]: You will need to dynamically check the existence of the method, like done here https://github.com/SkriptLang/Skript/blob/4f0fd027bfd6340f3966e1daf57c6189039f08b3/src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java#L365

**Comment by DeltaRays** [comment]
For me it states that the Line endings are CRLF so I'm not sure about that

File: src/main/java/ch/njol/skript/events/SimpleEvents.java [hunk 2]
```diff
@@ -110,414 +110,414 @@
 public class SimpleEvents {
 	static {
 		Skript.registerEvent("Can Build Check", SimpleEvent.class, BlockCanBuildEvent.class, "[block] can build check")
-				.description("Called when a player rightclicks on a block while holding a block or a placeable item. You can either cancel the event to prevent the block from being built, or uncancel it to allow it.",
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/events/SimpleEvents.java:113]: You need to resolve your tab deletion for this class, it's making the diff hard to read.

File: src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java [hunk 3]
```diff
@@ -440,6 +440,14 @@ public DamageCause get(final EntityDeathEvent e) {
 			}
 		}, 0);
 		// ProjectileHitEvent
+		// ProjectileHitEvent#getPlayer was added in 1.11
+		if(Skript.methodExists(ProjectileHitEvent.class, "getHitBlock")) EventValues.registerEventValue(ProjectileHitEvent.class, Block.class, new Getter<Block, ProjectileHitEvent>() {
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/classes/data/BukkitEventValues.java:444]: Please move ``EventValues.registerEventValue`` onto the next line.

**Comment by DeltaRays** [comment]
Since I'm causing more issues with these updates I'll open a new pull request

**Comment by DeltaRays** [comment]
Closing in favor of #3190

**Pull request closed without merging**